ptracer/context_helper_linux_test.go

```diff
@@ -0,0 +1,263 @@
+			addrAlign: uintptr(pageSize - 50),
+			wantErr:   false,
+		},
+		{
+			name:      "large_buffer_unaligned",
+			buffSize:  pageSize * 2,
+			addrAlign: 123,
+			wantErr:   false,
+		},
+		{
+			name:      "buffer_smaller_than_to_boundary",
+			buffSize:  10,
+			addrAlign: uintptr(pageSize - 100), // distance to page boundary is 100 bytes, but buffer is only 10 bytes
+			wantErr:   false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			buff := make([]byte, tc.buffSize)
+			
+			// get a readable memory address
+			maps, err := os.ReadFile(fmt.Sprintf("/proc/%d/maps", pid))
+			if err != nil {
+				t.Fatalf("Failed to read process maps: %v", err)
+			}
+
+			var baseAddr uintptr
+			for _, line := range bytes.Split(maps, []byte{'\n'}) {
+				if len(line) == 0 {
+					continue
+				}
+				if bytes.Contains(line, []byte("r-x")) {
+					var start uint64
+					fmt.Sscanf(string(line), "%x-", &start)
+					baseAddr = uintptr(start)
+					break
+				}
+			}
+
+			if baseAddr == 0 {
+				t.Fatal("Failed to find readable memory region")
+			}
+
+			// use test case specified alignment offset
+			testAddr := baseAddr + tc.addrAlign
+
+			// record buffer content before reading
+			originalBuff := make([]byte, len(buff))
+			copy(originalBuff, buff)
+
+			err = vmReadStr(pid, testAddr, buff)
+			if (err != nil) != tc.wantErr {
+				t.Errorf("vmReadStr() error = %v, wantErr %v", err, tc.wantErr)
+			}
+
+			// verify if actual reading occurred
+			if !bytes.Equal(buff, originalBuff) {
+				// at least some data was read
+				t.Logf("Data was read successfully for case: %s", tc.name)
+			}
+
+			// special case: check buffer size smaller than distance to boundary
+			if tc.name == "buffer_smaller_than_to_boundary" {
+				distToBoundary := pageSize - int(testAddr%uintptr(pageSize))
+				if distToBoundary > len(buff) {
+					t.Logf("Verified buffer handling when smaller than distance to boundary: dist=%d, buff=%d",
+						distToBoundary, len(buff))
+				}
+			}
+		})
+	}
+}
+
+// TestSliceBehavior tests slice behavior
+func TestSliceBehavior(t *testing.T) {
+	tests := []struct {
+		name      string
+		buffSize  int
+		nextRead  int
+		expected  int
+	}{
+		{
+			name:     "small_buffer_large_read",
+			buffSize: 10,
+			nextRead: 4096,
+			expected: 10,  // must be limited to buffer size
+		},
+		{
+			name:     "large_buffer_small_read",
+			buffSize: 8192,
+			nextRead: 4096,
+			expected: 4096,  // can use full read amount
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			buff := make([]byte, tt.buffSize)
+			// safely calculate actual read amount
+			actualRead := tt.nextRead
+			if tt.buffSize < actualRead {
+				actualRead = tt.buffSize
+			}
+			slice := buff[:actualRead]
+			
+			if len(slice) != tt.expected {
+				t.Errorf("Expected slice len %d, got %d", tt.expected, len(slice))
+			}
+		})
+	}
+}
+
+func min(a, b int) int {
+	if a < b {
+		return a
+	}
+	return b
+}
```

Comment:
`min` is already a built in function since go1.21, so we don't need to define it anymore.